### PR TITLE
HAWKULAR-596 : Improved counter/rate data at GC

### PR DESCRIPTION
**Pull request by ammendonca**
Populating missing data (for start and in the middle) with 0 values so
that time range and generated buckets are good.

Also upgraded to Hawkular Metrics 0.6.0.Final

### CODE REVIEW

File: console/src/main/scripts/plugins/metrics/ts/app-details/appServerJvmDetails.ts [hunk 1]
```diff
@@ -143,21 +143,57 @@ module HawkularMetrics {
     }
 
     private formatCounterChartOutput(response, buckets = 60):IChartDataPoint[] {
```
Comment (mtho11) [inline console/src/main/scripts/plugins/metrics/ts/app-details/appServerJvmDetails.ts]: Can this go in the MetricsService as it should be reusable for all counters
https://github.com/hawkular/hawkular/blob/ee54b476d14c039f5bb5df908ecab44ce547ae31/console/src/main/scripts/plugins/metrics/ts/services/metricsService.ts#L82-82
Also, can you add a retrieveCounterMetrics method.
This can be in a separate commit.

Comment (ammendonca) [inline console/src/main/scripts/plugins/metrics/ts/app-details/appServerJvmDetails.ts]: This is something supposed to be (very) temporary, because once metrics have support for buckets in counters most of it will go away, the remaining will go with the ability of charts to handle metrics unmodified [bucket] data, as per https://issues.jboss.org/browse/HAWKULAR-643.

Still, this can go in the MetricsService, don't think there's anything specific to it that'd block it. Moving.

**Comment by mtho11** [comment]
@ammendonca can you rebase again?

**Comment by ammendonca** [comment]
New PR https://github.com/hawkular/hawkular/pull/504. Closing this one.
